services/wallet/thirdparty/opensea/client.go

```diff
@@ -167,28 +189,53 @@ func (o *Client) FetchAllCollectionsByOwner(owner common.Address) ([]Collection,
 			break
 		}
 	}
-	o.IsConnected = true
+	o.setConnected(true)
 	return collections, nil
 }
 
 func (o *Client) FetchAllAssetsByOwnerAndCollection(owner common.Address, collectionSlug string, limit int) ([]Asset, error) {
-	offset := 0
+	queryParams := url.Values{
+		"owner":      {owner.String()},
+		"collection": {collectionSlug},
+	}
+
+	return o.fetchAssets(queryParams, limit)
+}
+
+func (o *Client) FetchAssetsByNFTUniqueID(uniqueIDs []NFTUniqueID, limit int) ([]Asset, error) {
+	queryParams := url.Values{}
+
+	for _, uniqueID := range uniqueIDs {
+		queryParams.Add("token_ids", uniqueID.TokenID.String())
+		queryParams.Add("asset_contract_addresses", uniqueID.ContractAddress.String())
+	}
+
+	return o.fetchAssets(queryParams, limit)
+}
+
+func (o *Client) fetchAssets(queryParams url.Values, limit int) ([]Asset, error) {
 	var assets []Asset
-	o.IsConnectedLock.Lock()
-	defer o.IsConnectedLock.Unlock()
-	o.LastCheckedAt = time.Now().Unix()
+
+	queryParams["limit"] = []string{strconv.Itoa(AssetLimit)}
+	var currentCursor string
 	for {
-		url := fmt.Sprintf("%s/assets?owner=%s&collection=%s&offset=%d&limit=%d", o.url, owner, collectionSlug, offset, AssetLimit)
+		if len(currentCursor) > 0 {
+			queryParams["cursor"] = []string{currentCursor}
+		} else {
+			queryParams["cursor"] = []string{`""`}
```

Comment:
@dlipicar is this what you really want? To set an array with a single element which is an empty string to `queryParams["cursor"]`? I thought maybe it's better to proceed without else block here, but if we need to have this (because of final url), then instead of `if` condition a single line `queryParams["cursor"] = []string{currentCursor}` is enough.

Reply:
If I set it to and empty string, the request will look something like:
`https://api.opensea.io/api/v1/assets?asset_contract_addresses=0xBC4CA0EdA7647A8aB7C2061c2E118A18a936f13D&**cursor=**&limit=50&token_ids=1273` which OpenSea doesn't like. So my options would be to remove the parameter from the `queryParams ` map, or do what I did (which OpenSea accepts). The first option seems cleaner now that I think about it... I'll make the change.

Reply:
Done, parameter is only added after first query, if next cursor is empty we simply exit the loop since there are no more collectibles to fetch